glambie/data/read_and_edit_bucket_csv_files.py

```diff
@@ -0,0 +1,343 @@
+                    interpolated_data_frame['date'] = interpolated_data_frame['start_date']
+                    interpolated_data_frame['date_fractional'] = start_date_fractional
+                    interpolated_data_frame['region_id'] = [submission_data_frame['region_id'][0]
+                                                            for i in range(len(interpolated_data_frame))]
+
+                    submission_data_frame = interpolated_data_frame.copy()
+
+                else:
+                    # If it is not a gravimetry file, then the data it contains is invalid.
+                    file_check_dataframe.loc[
+                        file_check_dataframe.local_filepath.__eq__(file.local_filepath),
+                        'reason_for_edit'] = 'Non-gravimetry file with gaps in data - inspect manually'
+
+            # Save out to same path as original file
+            submission_data_frame.to_csv(file.local_filepath)
+
+        if not file.nodata_check_satisfied:
+            submission_data_frame = pd.read_csv(file.local_filepath)
+
+            if 'm' in submission_data_frame.unit.values[0]:
+                # delete rows with change values > +/-100 - these numbers need some thought
+                submission_data_frame.drop(submission_data_frame[abs(
+                    submission_data_frame.glacier_change_observed) > 100].index, inplace=True)
+            elif 'Gt' in submission_data_frame.unit.values[0]:
+                # delete rows with change values > +/-10000
+                submission_data_frame.drop(submission_data_frame[abs(
+                    submission_data_frame.glacier_change_observed) > 10000].index, inplace=True)
+
+            submission_data_frame.to_csv(file.local_filepath)
+            # Record that the file has been edited
+            file_check_dataframe.loc[file_check_dataframe.local_filepath.__eq__(file.local_filepath),
+                                     'reason_for_edit'] = 'Random no data value was used'
+
+    return file_check_dataframe
+
+
+def main(upload: bool = False):
+
+    local_path = '/path/to/local/folder'
+    # If you want to download files for a specific region, set the region_prefix and supply here
+    downloaded_files = download_csvs_from_bucket(local_path, region_prefix=None)
+    file_check_results_dataframe = generate_results_dataframe(downloaded_files, local_path)
+    record_of_edits_dataframe = edit_local_copies_of_glambie_csvs(file_check_results_dataframe, local_path)
+
+    record_of_edits_dataframe.to_csv(os.path.join(local_path, 'record_of_edited_files.csv'))
+
+    # Final step that needs to be implemented here is to upload the edited files into the bucket, after copying the
+    # original version of the file into an archive folder
+    if upload:
+        upload_edited_files_to_bucket(record_of_edits_dataframe)
+
+
+if __name__ == "__main__":
+    upload_after_editing = False
+    main(upload_after_editing)
```

Comment:
Given this construct, this is sufficient/cleaner:
```suggestion
    main(upload=False)
```

Reply:
Done